Source/LocalNetAppChat/LocalNetAppChat.Bot/PluginProcessor/Plugins/PowerShellPlugin.cs

```diff
@@ -0,0 +1,62 @@
+﻿using LocalNetAppChat.Domain.Serverside;
+using LocalNetAppChat.Domain.Shared;
+using System.Management.Automation;
+
+namespace LocalNetAppChat.Bot.PluginProcessor.Plugins
+{
+    public class PowerShellPlugin : IPlugin
+    {
+        private readonly string _pluginsFolder = Directory.GetCurrentDirectory() + "..\\..\\my_scripts";
+
+        public PowerShellPlugin()
+        {
+
+        }
+
+        public string ExecuteCommand(string command)
+        {
+            if (CommandMessageTokenizer.IsCommandMessage(command)) return "Invalid Command!";
+
+            var rest = CommandMessageTokenizer.MessageWithoutCommandSignal(command);
+            var token = CommandMessageTokenizer.GetToken(ref rest);
+
+            if (token == "execps")
+            {
+                var scriptName = CommandMessageTokenizer.GetToken(ref rest);
+                if (!CheckIfScriptExists(scriptName)) return $"Script {scriptName} does not exist";
+                return ExecutePowerShellCommand(scriptName, rest);
+            }
+
+            throw new Exception("Invalid command syntax");
+        }
+
+        private bool CheckIfScriptExists(string scriptName)
+        {
+            string searchPattern = scriptName + ".ps1";
+            string[] fileNames = Directory.GetFiles(_pluginsFolder, searchPattern);
+
+            return fileNames.Length > 0;
+        }
+
+        private string ExecutePowerShellCommand( string scriptName, string parameters)
+        {
+            string scriptPath = _pluginsFolder + scriptName + ".ps1";
+            string[] scriptArgs = { parameters };
+
+            PowerShell ps = PowerShell.Create();
```

Comment:
I think it would be a bad idea to bind directly to powershell.

Reply:
Hey @stho32, 

What exactly do you mean? This is the Powershell plugin which will handle powershell commands, if I do not handle commands and write the logic for the execution in this plugin, where else ?

Reply:
Do not see powershell as powershell. Powershell is part of the .net eco system and the interface can be included directly in .net applications. But that is too close because powershell is not powershell. It can be the Framework-Powershell or the dotnet-core-powershell. The one bound to windows and the one not bound to windows. The 32bit version and the 64bit version that exist both within windows. Instead, see it as just another executable, just an external process.